// @nextnm/nestjs-mailgun/dist/nestjs-mailgun/services/relay/mailgun.service.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const testing_1 = require("@nestjs/testing");
const tokens_1 = require("../../../nestjs-mailgun/tokens/tokens");
const mailgun_service_1 = require("./mailgun.service");
describe('MailgunService', () => {
    let service;
    beforeEach(async () => {
        const module = await testing_1.Test.createTestingModule({
            providers: [
                mailgun_service_1.MailgunService,
                {
                    provide: tokens_1.MAILGUN_CONFIGURATION,
                    useValue: {
                        username: 'Nuno',
                        key: process.env.MAILGUN_KEY,
                    },
                },
            ],
        }).compile();
        service = module.get(mailgun_service_1.MailgunService);
    });
    it('should be defined', () => {
        expect(service).toBeDefined();
    });
    it('Send email', async () => {
        expect(await service.createEmail(process.env.MAILGUN_DOMAIN, {
            from: 'package@test.com',
            subject: 'TEST',
            to: 'benfica000@gmail.com',
            text: 'Test was successful',
        })).toBeDefined();
    });
    it.skip('Validate email', () => {
        return expect(service.validateEmail('postmaster@closeroffice.closerdev.com')).resolves.toBeDefined();
    });
});
